docs/custom_promise_types/backup.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+#
+# Backup custom promise type, uses cfengine.sh library located in same dir,
+# and https://github.com/Lex-2008/backup3
+#
+# Installation:
+# 1. Clone the above repo next to this file
+# 2. pick a directory to keep backups, preferrably on separate disk/partition,
+#    since backups are configured to fill up to 90% of disk space by default.
+#    We'll use /var/backups in this example.
+# 3. Run init.sh to initialise backup dir, like this:
+#    BACKUP_ROOT=/var/backups backup3/init.sh
+# 4. Use it in the policy like this:
+#      promise agent backup
+#      {
+#          interpreter => "/bin/bash";
+#          path => "$(sys.inputdir)/backup.sh";
+#      }
+#      bundle agent main
+#      {
+#        backup:
+#          "homes"
+#            from => "/home/",
+#            root => "/var/backups";
+#      }
+
+obligatory_attributes="from"
+optional_attributes="root"
+all_attributes_are_valid="no"
+
+do_evaluate() {
+    # Prepare environment for wrapped script
+    BACKUP_ROOT="$request_attribute_root"
+    BACKUP_KEEP_TMP="1"
+    run_this() {
+        run_rsync always "$request_promiser" "$request_attribute_from"
+    }
+
+    # call the script
+    . "$(dirname "$0")/backup3/backup.sh"
+
+    # Analyse its results
+    if [ ! -f "$BACKUP_TMP".files ]; then
+        response_result="not_kept"
+        return 0
+    fi
+    local filelist_size="$(stat -c %s "$BACKUP_TMP".files)"
+    if [ "$filelist_size" -gt 21 ]; then
```

Comment:
?

Reply:
indeed, it was a bit of magic. Replaced with explanation